source/src/GeometryUtil.cc

```diff
@@ -57,3 +56,36 @@ dd4hep::rec::LayeredCalorimeterData const* MarlinUtil::getLayeredCalorimeterData
 
   return theExtension;
 }
+
+
+template<class DetExtension>
+DetExtension* MarlinUtil::getDetData(const std::string& detName){
+    auto& detector = dd4hep::Detector::getInstance();
+    auto detElem = detector.detector(detName);
+    auto detData = detElem.extension<DetExtension>();
+    return detData;
+}
```

Comment:
This will have to be in the header file. Otherwise only the specializations that you use below will be available. If someone tries to get another detector extension via this method it will lead to a linker error, because the compiler will not have generated other specializations than the ones used below. If you put it into the header file the compiler always sees the full definition and can generate new specializations when needed.